nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/DeleteHBaseRow.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Tags({ "delete", "hbase" })
+@CapabilityDescription(
+        "Delete HBase records individually or in batches. The input can be a single row ID in the body, one ID per line, " +
+        "row IDs separated by commas or a combination of the two. ")
+public class DeleteHBaseRow extends AbstractDeleteHBase {
+    static final AllowableValue ROW_ID_BODY = new AllowableValue("body", "FlowFile content", "Get the row key(s) from the flowfile content.");
+    static final AllowableValue ROW_ID_ATTR = new AllowableValue("attr", "FlowFile attributes", "Get the row key from an expression language statement.");
+
+    static final PropertyDescriptor ROW_ID_LOCATION = new PropertyDescriptor.Builder()
+            .name("delete-hb-row-id-location")
+            .displayName("Row ID Location")
+            .description("The location of the row ID to use for building the delete. Can be from the content or an expression language statement.")
+            .required(true)
+            .defaultValue("body")
+            .allowableValues(ROW_ID_BODY, ROW_ID_ATTR)
+            .addValidator(Validator.VALID)
+            .build();
+
+    static final PropertyDescriptor FLOWFILE_FETCH_COUNT = new PropertyDescriptor.Builder()
+            .name("delete-hb-flowfile-fetch-count")
+            .displayName("Flowfile Fetch Count")
+            .description("The number of flowfiles to fetch per run.")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("5")
+            .expressionLanguageSupported(false)
+            .build();
+
+    static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("delete-hb-row-ff-count")
+            .displayName("Batch Size")
+            .description("The number of deletes to send per batch.")
+            .required(true)
+            .defaultValue("50")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(false)
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = super.getSupportedPropertyDescriptors();
+        properties.add(ROW_ID_LOCATION);
+        properties.add(FLOWFILE_FETCH_COUNT);
+        properties.add(BATCH_SIZE);
+
+        return properties;
+    }
+
+    @Override
+    protected void doDelete(ProcessContext context, ProcessSession session) throws Exception {
+        final int batchSize     = context.getProperty(BATCH_SIZE).asInteger();
```

Comment:
I think that the preferred syntax in the project is to have only one space before and after the `=` sign

Reply:
It might be, but I've never been told to change it before. If someone wants to make a real issue of it, I'll change it. Otherwise, I think it helps with readability in cases like this.